clients/snowflake/append.go

```diff
@@ -0,0 +1,60 @@
+package snowflake
+
+import (
+	"context"
+	"log/slog"
+
+	"github.com/artie-labs/transfer/lib/config/constants"
+	"github.com/artie-labs/transfer/lib/destination/ddl"
+	"github.com/artie-labs/transfer/lib/typing/columns"
+
+	"github.com/artie-labs/transfer/lib/optimization"
+)
+
+func (s *Store) Append(ctx context.Context, tableData *optimization.TableData) error {
+	err := s.append(tableData)
+	if IsAuthExpiredError(err) {
+		slog.Warn("authentication has expired, will reload the Snowflake store and retry appending", slog.Any("err", err))
+		s.reestablishConnection()
+		return s.Append(ctx, tableData)
```

Comment:
In the case that the auth is permanently expired this will recurse forever, should we add some type of max retries?

Reply:
We can add a retry, I'll leave a TODO. We need that for `Merge` too.

Reply:
We'll need to find a way to keep track of retry count since this function sig is used as part of the interface